### ISSUE TITLE
Create working possibility to map all scalar types but only explicitely defined pojo types

### ISSUE THREAD

**Issue by Xnyle** [mapstruct/mapstruct#2497]
I need to have full control over what properties are getting mapped but based on type, not on name.

In essence I'd like to always map all primitive types and Pojo types only if explicitely defined.

I failed to do that with the current implementation.

To disable all internal mappings and conversions I used  mapping control MAPPING_METHOD, I furthermore set disableSubMappingMethodsGeneration = true on the mapper.

My idea was that I could then create mapping interfaces for scalar types and for the pojo references i really want add @Mapping annotations.

first problem is that a String mapScalar(String in) interface always creates an empty String implementation, but ok i can write my own implemtation vua abstract mapper class.

The real problem is that if I then have undefinied mappings (by excluding everything but mapping methods), i get errors about missing mappings. I can influence if I get a warning, error, or ignore about that via unmappedTargetPolicy but even if i set that to ignore, i still get an error telling me i should implement a mapper method (for a property i just don't want to get mapped)

I could use @BeanMapping(ignoreByDefault = true) but then everything is excluded, even scalar types and I would have to define all scalar properties for all (lots of) pojo types explicitely. And then I'm back at my original problem that there is no solution to automatically include all scalar types.

To summarize I need a possibility to
1. automatically map all scalar types 
2. descend into non scalar types only if explicitely defined

**Comment by sjaakd** [comment]
Reading your story it is not exactly clear what you want. Have you looked at qualifiers? 

Remember MapStruct maps in a limited number of types:
1. bean to bean
2. enum to enum/string 
3. map to map
4. iterable to iterable
5. stream to stream.
...
It cannot generate mapping methods for scalar or primitive types, and its neither intended to do so
